2023-08-30 15:12:58,634 - worker_process - INFO - Worker_process started.
2023-08-30 15:12:58,888 - Cell - INFO - site-1.f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d: created backbone internal connector to tcp://localhost:29726 on parent
2023-08-30 15:12:58,912 - Cell - INFO - site-1.f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d: created backbone external connector to grpc://192.168.100.3:8102
2023-08-30 15:12:58,913 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00001 ACTIVE tcp://localhost:29726] is starting
2023-08-30 15:12:58,913 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00002 ACTIVE grpc://192.168.100.3:8102] is starting
2023-08-30 15:12:58,914 - FederatedClient - INFO - Wait for client_runner to be created.
2023-08-30 15:12:58,915 - FederatedClient - INFO - Got client_runner after 0.0009381771087646484 seconds
2023-08-30 15:12:58,915 - FederatedClient - INFO - Got the new primary SP: grpc://192.168.100.3:8102
2023-08-30 15:12:58,923 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: client runner started
2023-08-30 15:12:58,949 - Communicator - INFO - Received from secure_project server  (1114733 Bytes). getTask: train time: 0.016783714294433594 seconds
2023-08-30 15:12:58,950 - FederatedClient - INFO - pull_task completed. Task name:train Status:True 
2023-08-30 15:12:58,950 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, peer=secure_project, peer_run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: got task assignment: name=train, id=d32044b9-c571-4637-b6be-2ddd6685c4b5
2023-08-30 15:12:58,950 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, peer=secure_project, peer_run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, task_name=train, task_id=d32044b9-c571-4637-b6be-2ddd6685c4b5]: invoking task executor LearnerExecutor
2023-08-30 15:12:58,950 - LearnerExecutor - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, peer=secure_project, peer_run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, task_name=train, task_id=d32044b9-c571-4637-b6be-2ddd6685c4b5]: Client trainer got task: train
2023-08-30 15:12:58,950 - MNISTLearner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, peer=secure_project, peer_run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, task_name=train, task_id=d32044b9-c571-4637-b6be-2ddd6685c4b5]: Client site-1 initialized at 
 /home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d/app_site-1 
 with args: Namespace(workspace='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/..', startup='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../startup', token='d2c82c93-15f6-4e9e-8318-e0e19a0309fe', ssid='ebc6125d-0a56-4688-9b08-355fe9e4d61a', job_id='f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d', client_name='site-1', sp_target='192.168.100.3:8102', parent_url='tcp://localhost:29726', fed_client='fed_client.json', set=['secure_train=true', 'uid=site-1', 'org=nvidia', 'config_folder=config', 'print_conf=True'], local_rank=0, train_config='config/config_train.json', client_config='config/config_fed_client.json', config_folder='config', env='config/environment.json')
2023-08-30 15:13:00,255 - ClientRunner - ERROR - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, peer=secure_project, peer_run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, task_name=train, task_id=d32044b9-c571-4637-b6be-2ddd6685c4b5]: Processing error from executor LearnerExecutor: ValueError: num_samples should be a positive integer value, but got num_samples=0
2023-08-30 15:13:00,256 - ClientRunner - ERROR - Traceback (most recent call last):
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/private/fed/client/client_runner.py", line 254, in _process_task
    reply = executor.execute(task.name, task.data, fl_ctx, self.task_abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 86, in execute
    return self.train(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 99, in train
    validate_result: Shareable = self.learner.validate(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 357, in validate
    self._create_datasets(fl_ctx)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 165, in _create_datasets
    self.train_loader = self.train_datamgr.get_loader(self.transform_train, batch_size=self.batch_size, shuffle=True, num_workers=self.num_workers, train=True)
  File "/home/aienode1/NVFlare/sound_cls/pt/utils/data_manager.py", line 36, in get_loader
    loader = data.DataLoader(dataset=dataset, shuffle=shuffle, batch_size=batch_size, num_workers=num_workers, drop_last=True, collate_fn=self.pad_seq)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 351, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore[arg-type]
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/sampler.py", line 107, in __init__
    raise ValueError("num_samples should be a positive integer "
ValueError: num_samples should be a positive integer value, but got num_samples=0

2023-08-30 15:13:00,257 - FederatedClient - INFO - Starting to push execute result.
2023-08-30 15:13:00,260 - Communicator - INFO -  SubmitUpdate size: 575 Bytes. time: 0.0035419464111328125 seconds
2023-08-30 15:13:00,261 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, peer=secure_project, peer_run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, task_name=train, task_id=d32044b9-c571-4637-b6be-2ddd6685c4b5]: result sent to server for task: name=train, id=d32044b9-c571-4637-b6be-2ddd6685c4b5
2023-08-30 15:13:00,871 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d, peer=secure_project, peer_run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: received aux request from Server to end current RUN
2023-08-30 15:13:00,871 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: ABORT (RUN) command received
2023-08-30 15:13:00,872 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: ABORT (RUN) requests end run events sequence
2023-08-30 15:13:00,872 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: ABOUT_TO_END_RUN fired
2023-08-30 15:13:00,873 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: END_RUN fired
2023-08-30 15:13:02,264 - ClientRunner - INFO - [identity=site-1, run=f5ac6418-a0bb-4e10-a0dd-5fbfa7e5725d]: run method requests end run events sequence
2023-08-30 15:13:02,280 - FederatedClient - INFO - Shutting down client run: site-1
2023-08-30 15:13:04,241 - MPM - INFO - MPM: Good Bye!
